ethdb/database.go

```diff
@@ -194,4 +199,5 @@ type Database interface {
 	Snapshotter
 	io.Closer
 	WasmDataBaseRetriever
+	WasmTargetsRetriever
```

Comment:
I moved the WasmTargets() method to WasmDataBaseRetriever. 

Would it make sense to instead of having `wasmdb` and `wasmTag` in `cachingDB` to have:
```
func (db *cachingDB) WasmStore() ethdb.KeyValueStore {
    wasmdb, _ := wasmDatabsaseRetriever.WasmDatabase()
    return wasmdb
}

func (db *cachingDB) WasmCacheTag() uint32 {
    _, wasmTag := wasmDatabsaseRetriever.WasmDatabase()
    return wasmTag
}
```
?